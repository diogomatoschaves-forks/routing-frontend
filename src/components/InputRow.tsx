import React, { Component } from 'react'
import { Box, StyledIcon, ColoredDiv } from '../styledComponents'
import ControlledInput from './ControlledInput'
import { UpdatePoint, Coords } from '../types'
import { NORMAL_INPUT, FOCUSED_INPUT } from '../utils/colours'


interface Props {
  rowKey: string,
  index: number,
  coords: Coords,
  placeholder: string,
  iconName: string | any,
  updatePoint: UpdatePoint
}

const diameter = 38

class InputRow extends Component<Props, any> {

  constructor(props: Props) {
    super(props)
    this.state = {
      color: NORMAL_INPUT
    }
  }

  updateColor = () => {
    this.setState((state: any) => ({ color: state.color === NORMAL_INPUT ? FOCUSED_INPUT : NORMAL_INPUT }))
  }

  public render() {

    const { iconName, rowKey, index, coords, updatePoint, placeholder } = this.props
    const { color } = this.state

    return (
      <Box direction="row" justify="space-around" padding="10px 0">
        <ColoredDiv circle diameter={diameter} color={color} margin="0 10px 0 0" position="relative">
          <Box height="100%" padding={`${Math.round(diameter / 2)}px`}>
            <StyledIcon 
              fontSize={'25px'}
              height={'20px'}
              padding="0 0 0 0" 
              overridecolor={'white'} 
              name={iconName}
              position="absolute"
            />
          </Box>
        </ColoredDiv>
        <ControlledInput 
          rowKey={rowKey}
          index={index}
          coords={coords}
          updatePoint={updatePoint}
          updateColor={this.updateColor}
          placeholder={placeholder}
        />
      </Box>
    )
  }
}

export default InputRow