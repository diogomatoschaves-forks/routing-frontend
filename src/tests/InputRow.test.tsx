import React from 'react'
import TestRenderer from 'react-test-renderer'
import InputRow from '../components/InputRow'

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  GeolocateControl: jest.fn(),
  Map: jest.fn(() => ({
    addControl: jest.fn(),
    on: jest.fn(),
    remove: jest.fn()
  })),
  NavigationControl: jest.fn()
}));

it('expect to render Panel component', () => {

  const item = { 
    name: 'start', 
    marker: 'map marker alternate',
    placeholder: 'Origin',
    point: null
  }

  const testInstance = TestRenderer.create(
    <InputRow
      rowKey={item.name}
      index={0}
      placeholder={item.placeholder}
      iconName={item.marker} 
      updatePoint={jest.fn()}
    />
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
}) 
