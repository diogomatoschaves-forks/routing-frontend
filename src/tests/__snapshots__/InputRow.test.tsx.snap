// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Panel component 1`] = `
<div
  className="sc-bdVaJa ecsjM"
  direction="row"
>
  <div
    className="sc-bwzfXH gXAlSF"
    color="#007a93"
  >
    <div
      className="sc-bdVaJa gqMWUP"
      height="100%"
    >
      <i
        aria-hidden="true"
        className="map marker alternate icon sc-htpNat iIqnlv"
        fontSize="25px"
        height="20px"
        onClick={[Function]}
        overridecolor="white"
        padding="0 0 0 0"
        position="absolute"
      />
    </div>
  </div>
  <div
    className="ui fluid input sc-EHOje kbRwGm"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Origin"
      type="text"
      value=""
    />
  </div>
</div>
`;
